lua-api-crates/table-funcs/src/lib.rs

```diff
@@ -0,0 +1,254 @@
+                    let inner_tbl = deep_extend(
+                        lua,
+                        (vec![tbl.get(key.clone())?, t], Some(ConflictMode::Keep)),
+                    )?;
+                    tbl.set(key, inner_tbl)?;
+                }
+            }
+        }
+        // default behavior is to keep last set value
+        Some(ConflictMode::Force) | None => {
+            for (key, value) in tbl_vec {
+                if !tbl.contains_key(key.clone())? {
+                    tbl.set(key, value)?;
+                } else if let LuaValue::Table(t) = value {
+                    let inner_tbl = deep_extend(
+                        lua,
+                        (vec![tbl.get(key.clone())?, t], Some(ConflictMode::Force)),
+                    )?;
+                    tbl.set(key, inner_tbl)?;
+                } else {
+                    tbl.set(key, value)?;
+                }
+            }
+        }
+        Some(ConflictMode::Error) => {
+            for (key, value) in tbl_vec {
+                if !tbl.contains_key(key.clone())? {
+                    tbl.set(key, value)?;
+                } else if let LuaValue::Table(t) = value {
+                    let inner_tbl = deep_extend(
+                        lua,
+                        (vec![tbl.get(key.clone())?, t], Some(ConflictMode::Keep)),
+                    )?;
+                    tbl.set(key, inner_tbl)?;
+                } else {
+                    return Err(mlua::Error::runtime(format!(
+                        "The key {} is in more than one of the tables.",
+                        key.to_string()?
+                    )));
+                }
+            }
+        }
+    };
+
+    Ok(tbl)
+}
+
+fn clone<'lua>(lua: &'lua Lua, table: Table<'lua>) -> mlua::Result<Table<'lua>> {
+    let table_len = table.clone().pairs::<LuaValue, LuaValue>().count();
+    let res: Table<'lua> = lua.create_table_with_capacity(0, table_len)?;
+
+    for pair in table.pairs::<LuaValue, LuaValue>() {
+        let (key, value) = pair?;
+        if let LuaValue::Table(tbl) = value {
+            let inner_res = clone(lua, tbl)?;
+            res.set(key, inner_res)?;
+        } else {
+            res.set(key, value)?;
+        }
+    }
+    Ok(res)
+}
+
+fn flatten<'lua>(lua: &'lua Lua, arrays: Vec<LuaValue<'lua>>) -> mlua::Result<Vec<LuaValue<'lua>>> {
+    let mut flat_vec: Vec<LuaValue> = vec![];
+    for item in arrays {
+        match item {
+            LuaValue::Table(tbl) => {
+                let tbl_as_vec = tbl.sequence_values().filter_map(|x| x.ok()).collect();
+                let flat = flatten(lua, tbl_as_vec)?;
+                for j in flat {
+                    flat_vec.push(j);
+                }
+            }
+            LuaValue::Nil => (),
+            LuaValue::Thread(_) => (),
+            LuaValue::Error(err) => {
+                return Err(err);
+            }
+            other => {
+                flat_vec.push(other);
+            }
```

Comment:
I think skipping `nil` makes sense (although: given that lua treats `nil` as not assigned, I'd be surprised if we ever saw one as a value here), but all other types should just be flattened without caring about the value?
In particular, I wonder why threads are stripped out here; they are valid first class values.
```suggestion
            other => {
                flat_vec.push(other);
            }
```

Reply:
Oh, I didn't know that threads were valid values here (I think I misunderstood them). I will update it.

Reply:
[coroutine.create](https://www.lua.org/manual/5.3/manual.html#pdf-coroutine.create) returns a thread, for example.  While I don't think anyone is directly using this on wezterm today, I think there are some libraries out there that use coroutines to implement some generator style patterns